### DISCUSSION TITLE
--generation-codec=ascii missing ?

### DISCUSSION THREAD

**Discussion by dbire** [schemathesis/schemathesis#1937] [Q&A]
Hello, I was trying to use this option : "st run --generation-codec=ascii ..." as stated in the documentation but I get the following error:
"rror: No such option: --generation-codec (Possible options: --data-generation-method, --operation-id)"

Is it a bug ?

Is there a workaround ?

Thanks by advance

**Comment by Stranger6667** [comment]
What version of Schemathesis are you using? it is available starting from 3.22

**Reply by dbire** [reply]
Hello, thank you for super fast reply. My bad I was using docker **stable** image (3.20.2), which confused me because the documentation is labelled **stable** as well.
Nevertheless, I used --generation-codec=ascii with the 3.22.1 but I realized it was not what I expected. 
Actually the string generator creates a mix of encoded strings like ""\u00e8\u00c7\u0090\udb8e\udcb6\u00d9\u00e8<\ud936\udfb6\u00a9\udaf6\udec9\u00af" while I would rather use strings like "jkjlkqj!:lkdj" for two reasons: 1) it is more readable 2) it seems that I received 400s because I exceed the maximum length of the string because of the encoding each character generated takes 5 times and is counted as 5 characters.

Is there a way to handle this situation ? Thanks by advance :-)

**Reply by Stranger6667** [reply]
> Hello, thank you for super fast reply. My bad I was using docker stable image (3.20.2), which confused me because the documentation is labelled stable as well.

Docker images are updated every release, so I guess that happened because of local caching.

> \u00e8\u00c7\u0090\udb8e\udcb6\u00d9\u00e8<\ud936\udfb6\u00a9\udaf6\udec9\u00af

I have a few questions:
- What command did you use to generate it?
- What part of the request is it? Is it some part of the body payload? I am interested in how this data is serialized before being sent over the network

**Comment by dbire** [comment]
Sorry my bad again, I did not think about clearing my stable image.

I am using rhe following command:

docker run --network=my-network \
        -e SCHEMATHESIS_HOOKS=tests.extensions \
        -v /myspecs:/specs \
        -v ${PWD}:/app \
        schemathesis/schemathesis:3.22.1 run  /specs/api/v4/users.yaml \
        --code-sample-style curl \
        --workers auto \
        --checks all \
        --base-url http://users-api/api/v4 \
        --header "X-Api-Key: -----------" \
        --validate-schema=true \
        --endpoint ^/api/v4/users$ \
        --method POST \
        --hypothesis-max-examples 100 \
        --hypothesis-database None \
        --cassette-path cassette.yml

The string I mentionned is part of the body payload

here is an example of the body generated:

 {"firstName": "4`\u00d3]", "lastName": "4`\u00d3]", "password": "o\u00f6_\u00d4\udb1b\udd71av\u00e6l.#\ud9a4\udf28\u00f8!\uda11\ude57\u00bc\u001c%\u00e8\u00d6", "username": "\u0007B\u0007\u00a9b\u000bBX\u008c\u00bd\udb9c\ude30\uda52\udcd7a\u00c6\u00b9-"}
